### PR TITLE
Remove duplicate data classes

### PR DESCRIPTION
**Pull request by bartekryba**

### CODE REVIEW

File: lib/src/main/kotlin/com/swmansion/starknet/data/responses/Transaction.kt [hunk 1]
```diff
@@ -1,15 +1,19 @@
 package com.swmansion.starknet.data.responses
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/data/responses/Transaction.kt:1]: Why should move those types to data/Transactions if we want to use them for both responses and requests.

**Review by cptartur — COMMENTED** [review]
I like the changes introduced here, however I'm concerned about other thing. Account.sign returns a InvokeTransaction but how can this transaction can be then used in Provider. Perhaps we should modify Provider interface or add some helper methods so signed transactions can then be easily sent.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/Transactions.kt [hunk 2]
```diff
@@ -78,6 +70,9 @@ class BlockHashOrTagSerializer() : KSerializer<BlockHashOrTag> {
     }
 }
 
+class InvalidContractException(missingKey: String) :
+    Exception("Attempted to parse an invalid contract. Missing key: $missingKey")
+
```
[the remark this answers is not in the document]
Comment (bartekryba) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/Transactions.kt:75]: What about adding this as internal class inside ContractDefinition class?

File: lib/src/main/kotlin/com/swmansion/starknet/data/responses/Transaction.kt [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.json.JsonNames
 
-enum class TransactionType {
-    DECLARE, DEPLOY, INVOKE
+enum class TransactionType(val txPrefix: Felt) {
+    DECLARE(Felt.fromHex("0x6465636c617265")), // encodeShortString('declare'),
+    DEPLOY(Felt.fromHex("0x6465706c6f79")), // encodeShortString('deploy'),
+    INVOKE(Felt.fromHex("0x696e766f6b65")), // encodeShortString('invoke'),
```
[the remark this answers is not in the document]
Comment (bartekryba) [reply lib/src/main/kotlin/com/swmansion/starknet/data/responses/Transaction.kt:16]: These are the values used to calculate transaction hash. https://docs.starknet.io/docs/Blocks/transactions

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/Transactions.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/Transactions.kt]: I personally prefer having either separate files or classes in the same file. If you think that having it as an internal class is better though, I can accept that.

File: lib/src/main/kotlin/com/swmansion/starknet/data/responses/Transaction.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/responses/Transaction.kt:16]: Thanks 👍 , we should keep these if that's the case.

**Review by Solpatium — APPROVED** [review]

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionPayload.kt [hunk 4]
```diff
@@ -9,12 +9,12 @@ data class InvokeFunctionPayload(
...
 
-    val signature: Signature?,
+    val signature: Signature,
 
     @SerialName("max_fee")
-    val maxFee: Felt?,
+    val maxFee: Felt,
 
-    val version: Felt?,
+    val version: Felt,
```
Comment (cptartur) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionPayload.kt:17]: Can we define serial names here just in case?

**Review by cptartur — APPROVED** [review]
Just a single comment otherwise approved

**Pull request merged**